dataline-workers/src/main/java/io/dataline/workers/protocol/singer/SingerMessageTracker.java

```diff
@@ -51,7 +52,7 @@ public void accept(SingerMessage record) {
     if (record.getType().equals(SingerMessage.Type.STATE)) {
       final State state = new State();
       state.setConnectionId(connectionId);
-      state.setState(record);
+      state.setStateJson(record.getRecord());
```

Comment:
I think there as a bug here. @cgardens please confirm because the behavior is changing

Reply:
I actually think the original version is correct. The goal of our state object is to blackbox the entire state concept for an integration. in singers case the when the message is type state we want to just store that whole message as a blob. 

you can make an argument that we should do `record.getValue` since internal to the singer state file, the `value` field is where the actual state is stored. i'm inclined to just store the whole thing, but either way would work. 

i'm pretty sure `record.getRecord()` isn't right. `getRecord` will return a non null value if the message time is `RECORD`.

Reply:
also i'm writing a test for this class this afternoon so that it will be more clear in the future 😄